library/std/src/panicking.rs

```diff
@@ -288,7 +311,7 @@ pub mod panic_count {
     #[inline(never)]
     #[cold]
     fn is_zero_slow_path() -> bool {
-        LOCAL_PANIC_COUNT.with(|c| c.get() == 0)
+        get() == 0
```

Comment:
This would result in `std::thread::panicking()` always returning true when 'always abort' mode has been set.

Reply:
> This would result in `std::thread::panicking()` always returning true when 'always abort' mode has been set.

You are right.  I should have done the more typesafe thing with having `increase` return a tuple as you suggested.  That would have caught this.  The branch has become too much of a mess with this flail and I will force push it when I've fixed things and have a comprehensible story.